### ISSUE TITLE
Refactor the string_to_colour method to allow for transparent module backgrounds

### ISSUE THREAD

**Issue by bepzi** [ben01189998819991197253/contrail#12]
This will likely involve changing the method signature of `string_to_colour` from

```rust
fn string_to_colour(s: String) -> Colour
```

to

```rust
fn string_to_colour(s: String) -> Option<Colour>
```

If the user passes "none" or "transparent", then this function should return `None`. The callers of this function should then know to continue without any color at all, perhaps with an empty `Style::default()` instead.

**Comment by bepzi** [comment]
This might actually require a complete overhaul of the way we get the background, foreground, and text styles from the user. It would make more sense if the config file looked like this:

```toml
[global]
modules = ["modules", "more_modules"]  # etc

[global.style]
foreground = "bright_white"
background = "blue"
property = "normal"

[modules.more_modules]
output = "blah"

[modules.more_modules.style]
background = "green"
property = "bold"
# foreground inherits "bright_white" from global
```

because then it would be easier to pass/inherit foregrounds and backgrounds, and we could also do neat things like `style_error` instead of `bg_error` in the exit_code and prompt modules.

**Comment by bepzi** [comment]
Closed because this is being addressed in the v0.2 rewrite branch
